python/ray/air/_internal/torch_utils.py

```diff
@@ -228,3 +228,131 @@ def contains_tensor(obj):
             if contains_tensor(v):
                 return True
     return False
+
+
+# From PyTorch:
```

Comment:
can we remove this copyright? the new function is pretty different from the original.